typing/typecore.ml

```diff
@@ -5445,9 +5445,9 @@ let type_clash_of_trace trace =
 let report_literal_type_constraint expected_type const =
   let const_str = match const with
     | Const_int n -> Some (Int.to_string n)
-    | Const_int32 n -> Some (Int32.to_string n)
-    | Const_int64 n -> Some (Int64.to_string n)
-    | Const_nativeint n -> Some (Nativeint.to_string n)
+    | Const_int32 (_, s) -> Some s
+    | Const_int64 (_, s) -> Some s
+    | Const_nativeint (_, s) -> Some s
```

Comment:
This is the code that determines how the integers are rendered in the type hint.